tree/dataframe/inc/ROOT/RDF/RColumnRegister.hxx

```diff
@@ -35,7 +36,36 @@ namespace RDF {
 
 namespace RDFDetail = ROOT::Detail::RDF;
 
+class RDefineReader;
 class RVariationBase;
+class RVariationReader;
+
+/// A helper type that keeps track of RDefine objects and their corresponding RDefineReaders.
+class RDefineAndReaders {
+   using RDefineBase = RDFDetail::RDefineBase;
+
+   std::shared_ptr<RDefineBase> fDefine;
+   // Column readers per variation (in the map) per slot (in the vector).
+   std::vector<std::unordered_map<std::string, std::shared_ptr<RDefineReader>>> fReadersPerVariation;
+
+public:
+   RDefineAndReaders(std::shared_ptr<RDefineBase> define, unsigned int nSlots);
+   RDefineBase *GetDefine() const { return fDefine.get(); }
+   std::shared_ptr<RDefineReader>
+   GetReader(unsigned int slot, const std::string &variationName, const std::type_info &);
+};
+
+class RVariationAndReaders {
```

Comment:
If you like that change above I'd do the same here then:
```suggestion
class RVariationWithReaders {
```

Reply:
done, thanks